vm/actor/src/builtin/miner/mod.rs

```diff
@@ -415,16 +613,38 @@ impl Actor {
         BS: BlockStore,
         RT: Runtime<BS>,
     {
-        if params.expiration <= rt.curr_epoch() {
+        if !check_supported_proof_types(params.seal_proof) {
+            return Err(ActorError::new(
+                ExitCode::ErrIllegalArgument,
+                format!("unsupported seal proof type: {:?}", params.seal_proof),
+            ));
+        }
+
+        #[allow(clippy::absurd_extreme_comparisons)]
+        if params.sector_number > MAX_SECTOR_NUMBER {
             return Err(ActorError::new(
                 ExitCode::ErrIllegalArgument,
                 format!(
-                    "sector expiration {} must be after now {}",
-                    params.expiration,
-                    rt.curr_epoch()
+                    "sector number {} out of range 0..(2^63-1)",
+                    params.sector_number
                 ),
             ));
         }
+
+        if params.sealed_cid == Cid::default() {
```

Comment:
This does not actually match what Lotus is doing. They have a CID.Undef, which is different than our Cid::default. @austinabell can explain better.

Reply:
I always forget/confuse their CID Undef lol

Reply:
yeah, you should be able to safely ignore Cid.Undef checks, that is just a sanity check they have. Cid undef cannot be serialized, so it can never be an actual param. This is only added because these functions can be called directly in the go implementation